walletkeys/walletkeys.go

```diff
@@ -0,0 +1,138 @@
+package walletkeys
+
+import (
+	"fmt"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	bip39 "github.com/bitcoin-sv/go-sdk/compat/bip39"
+	chaincfg "github.com/bitcoin-sv/go-sdk/transaction/chaincfg"
+)
+
+// DefaultEntropy defines the default entropy (bit size) used for cryptographic purposes.
+// The value must be a multiple of 32 and within the inclusive range of {128, 256}.
+// It represents the default level of entropy for key generation or similar operations.
+const DefaultEntropy = 128
+
+// Keys represents a set of hierarchical deterministic (HD) keys,
+// including the extended private key (XPriv) and extended public key (XPub).
+type Keys struct {
+	xPriv string
+	xPub  string
+}
+
+// XPriv returns the HD extended private key as a string.
+func (k *Keys) XPriv() string { return k.xPriv }
+
+// XPub returns the HD extended public key as a string.
+func (k *Keys) XPub() string { return k.xPub }
+
+// KeysWithMnemonic extends the Keys struct by including the mnemonic phrase
+// used to generate the associated xPriv and XPub HD keys as strings.
+type KeysWithMnemonic struct {
+	keys     Keys
+	mnemonic string
+}
```

Comment:
I would rather prefer:
```suggestion
type KeysWithMnemonic struct {
	Keys
	mnemonic string
}
```